### DISCUSSION TITLE
Missing Basic Feature of building block of UI

### DISCUSSION THREAD

**Discussion by learncodingforweb** [slint-ui/slint#4750] [Q&A]
Unable to find following features to develop basic features.
1. Hide or Show component - no option is available
2. Pagination -  how can i add different pages in UI?
3. No date picker available
4. No Time Picker available

**Comment by ogoffart** [comment]
(I've converted the issue into a discussion because issue are more for bug report. It is better to use discussions for questions)

> 1. Hide or Show component - no option is available

To make individual component visible in .slint, there is a `visible` property that can be set to true or false.  You can also use the  `if (considion): FooBar { ... } `  construct

Of if you want to hide or show an entire compone,t you can use the `show()` or `hide()` funciton on slint::Window from native code.

> 2. Pagination - how can i add different pages in UI?

You can look for example at the printer demo example.

> 3. No date picker available
> 4. No Time Picker available

There is currently no builtin date picker indeed. You can try to make one yourself.
We have planed a date and time picker as part of the plan for our android port though: https://github.com/slint-ui/slint/issues/46
